server/controllers/favorites.js

```diff
@@ -1,39 +1,50 @@
 import models from '../models';
 
+const { Favourite, Recipe, User } = models;
+
 export default {
   addUserFavorites(req, res) {
-    return models.Recipe
-      .findById(req.params.recipeId)
-      .then((recipe) => {
-        if (!recipe) {
-          return res.status(400).json({
-            message: 'Recipe does not exist'
+    return Favourite
+      .findOne({
+        where: {
+          userId: req.decoded.user.id,
+          recipeId: req.params.recipeId
```

Comment:
Do `parseInt(req.params.recipeId, 10)`. That will save you some energy on Heroku.